### ISSUE TITLE
Package still displays in reference node after deleting <ItemGroup> node from csproj file

### ISSUE THREAD

**Issue by drewgillies** [NuGet/Home#4017]
Repro Steps:
1.	New a C# Console App (.Net Framwork).
2.	Add the following reference to the csproj file, before the final targets import line.
```
<ItemGroup>
        <PackageReference Include="Newtonsoft.Json">
            <Version>9.0.1</Version>
         </PackageReference>
</ItemGroup>
```

3.	Reload the project in VS and run solution restore, observe the reference node.
4.	Delete the above reference from the csproj file.
```
<ItemGroup> 
        <PackageReference Include="Newtonsoft.Json"> 
           <Version>9.0.1</Version> 
        </PackageReference> 
</ItemGroup>
```
5.	Reload the project in VS and run solution restore, observe the reference node again.

Expected:
‘Newtonsoft.Json’ should not display in reference node.

Actual:
Package still display in reference node
![image](https://cloud.githubusercontent.com/assets/18219758/20691964/a4f60ff6-b589-11e6-81ad-8493ca14142b.png)

**Comment by drewgillies** [comment]
An edge case as it pertains to manually removing references, but important as it exposes an invalidation hole in restore status.

**Comment by tmeschter** [comment]
After removing all the `PackageReference` items from the project, reloading, and restoring, is there still a project.assets.json file in the intermediates directory? If so, does it contain references to NewtonSoft.Json?

I suspect what is happening is that https://github.com/NuGet/NuGet.BuildTasks is picking up a stale project.assets.json file. If that is the case, I suggest that restore actively delete that file when a project has no `PackageReference` items.

**Comment by jainaashish** [comment]
Corresponding vsts issue# 299133 which will be fixed by ms build team.

**Comment by jainaashish** [comment]
Merged cb9b1a1559efac311b84aacc5f605004ad9e8562
